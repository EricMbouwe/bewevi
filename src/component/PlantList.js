import PropTypes from 'prop-types';
import PlantShortContainer from '../container/PlantShortContainer';

const PlantList = ({ plants }) => (
  <div className="plant-list main">
    {plants.map(plant => (
      <PlantShortContainer
        key={plant.id}
        plant={{
          id: plant.id,
          commonName: plant.common_name,
          scientificName: plant.scientific_name,
          imageUrl: plant.image_url,
        }}
      />
    ))}
  </div>

);

PlantList.propTypes = {
  plants: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    common_name: PropTypes.string.isRequired,
    scientificName: PropTypes.string.isRequired,
    image_url: PropTypes.string.isRequired,
  })).isRequired,
};

export default PlantList;
