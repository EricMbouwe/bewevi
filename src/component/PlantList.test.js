import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import PlantList from './PlantList';

jest.mock('../container/PlantShortContainer');

const plants = [
  {
    id: 1, common_name: 'foo1', scientific_name: 'bar1', image_url: 'foobar1',
  },
  {
    id: 2, common_name: 'foo2', scientific_name: 'bar2', image_url: 'foobar2',
  },
];

describe('<PlantList />', () => {
  it('renders other component', () => {
    render(
      <PlantList
        plants={plants}
      />,
    );
    screen.debug();
    expect(screen.getByText(/{"id":1,"commonName":"foo1","scientific_name":"bar1","imageUrl":"foobar1"}/i)).toBeInTheDocument();
    expect(screen.getByText(/{"id":2,"commonName":"foo2","scientific_name":"bar2","imageUrl":"foobar2"}/i)).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <PlantList
        plants={plants}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
