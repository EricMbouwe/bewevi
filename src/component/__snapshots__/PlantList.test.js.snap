// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlantList /> renders correctly 1`] = `
<div
  className="plant-list main"
>
  {"id":1,"commonName":"foo1","scientific_name":"bar1","imageUrl":"foobar1"}
  {"id":2,"commonName":"foo2","scientific_name":"bar2","imageUrl":"foobar2"}
</div>
`;
